src/unix/newlib/mod.rs

```diff
@@ -129,25 +129,6 @@ s! {
         pub tm_isdst: ::c_int,
     }
 
-    pub struct stat {
-        pub st_dev: ::dev_t,
-        pub st_ino: ::ino_t,
-        pub st_mode: ::mode_t,
-        pub st_nlink: ::nlink_t,
-        pub st_uid: ::uid_t,
-        pub st_gid: ::gid_t,
-        pub st_rdev: dev_t,
-        pub st_size: off_t,
-        pub st_atime: time_t,
-        pub st_spare1: ::c_long,
-        pub st_mtime: time_t,
-        pub st_spare2: ::c_long,
-        pub st_ctime: time_t,
-        pub st_spare3: ::c_long,
-        pub st_blksize: blksize_t,
-        pub st_blocks: blkcnt_t,
-        pub st_spare4: [::c_long; 2usize],
-    }
```

Comment:
This change was kinda invasive, I could swap it out for a few `#[cfg()]` instead, but unfortunately it seems `s!` is incompatible with `cfg_if!` (which would otherwise be more readable).

Reply:
I was trying to see if this is fixable, but it would require something like https://docs.rs/eager.